File: src/libs/ThirdParty.VBSuppliers/Properties/Jumbo/InjectedJumboSettings.vb
Imports Intuitive.Helpers.Extensions
Imports ThirdParty.Abstractions
Imports ThirdParty.Abstractions.Search.Settings
Imports ThirdParty.Abstractions.Support.SettingsSupport

Public Class InjectedJumboSettings
    Implements IJumboSettings

    Private ReadOnly configuration As ThirdPartyConfiguration

    Public Sub New(ByVal configuration As ThirdPartyConfiguration)
        Me.configuration = configuration
    End Sub

    Public ReadOnly Property AgencyCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.AgencyCode
        Get
            Return Get_Value("AgencyCode", configuration)
        End Get
    End Property

    Public ReadOnly Property BrandCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.BrandCode
        Get
            Return Get_Value("BrandCode", configuration)
        End Get
    End Property

    Public ReadOnly Property POS(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.POS
        Get
            Return Get_Value("POS", configuration)
        End Get
    End Property

    Public ReadOnly Property CommonsURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.CommonsURL
        Get
            Return Get_Value("CommonsURL", configuration)
        End Get
    End Property

    Public ReadOnly Property HotelBookingURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.HotelBookingURL
        Get
            Return Get_Value("HotelBookingURL", configuration)
        End Get
    End Property

    Public ReadOnly Property BasketHandlerURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.BasketHandlerURL
        Get
            Return Get_Value("BasketHandlerURL", configuration)
        End Get
    End Property

    Public ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IJumboSettings.AllowCancellations
        Get
            Return Get_Value("AllowCancellations", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property Language(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.Language
        Get
            Return Get_Value("Language", configuration)
        End Get
    End Property

    Public ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IJumboSettings.UseGZip
        Get
            Return Get_Value("UseGZIP", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, isMandatory As Boolean) As Integer Implements IJumboSettings.OffsetCancellationDays
        Get
            Return Get_Value("OffsetCancellationDays", configuration).ToSafeInt()
        End Get
    End Property

    Public ReadOnly Property NationalityBasedCredentials(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJumboSettings.NationalityBasedCredentials
        Get
            Return Get_Value("NationalityBasedCredentials", configuration)
        End Get
    End Property
End Class

File: src/libs/ThirdParty.CSSuppliers/Jumbo/InjectedJumboSettings.cs
namespace ThirdParty.CSSuppliers.Jumbo
{
    using Intuitive.Helpers.Extensions;
    using ThirdParty.Constants;
    using ThirdParty.Support;

    public class InjectedJumboSettings : SettingsBase, IJumboSettings
    {
        protected override string Source => ThirdParties.JUMBO;

        public string get_AgencyCode(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("AgencyCode", tpAttributeSearch);
        }

        public string get_BrandCode(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("BrandCode", tpAttributeSearch);
        }

        public string get_POS(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("POS", tpAttributeSearch);
        }

        public string get_CommonsURL(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("CommonsURL", tpAttributeSearch);
        }

        public string get_HotelBookingURL(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("HotelBookingURL", tpAttributeSearch);
        }

        public string get_BasketHandlerURL(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("BasketHandlerURL", tpAttributeSearch);
        }

        public bool get_AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("AllowCancellations", tpAttributeSearch).ToSafeBoolean();
        }

        public string get_Language(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("Language", tpAttributeSearch);
        }

        public bool get_UseGZip(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("UseGZIP", tpAttributeSearch).ToSafeBoolean();
        }

        public int get_OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool isMandatory)
        {
            return Get_Value("OffsetCancellationDays", tpAttributeSearch).ToSafeInt();
        }

        public string get_NationalityBasedCredentials(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("NationalityBasedCredentials", tpAttributeSearch);
        }
    }
}